src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -133,12 +133,20 @@ bool convert_precision(ov::pass::PassBase& pass,
         bool res = false;
         // Handle case with Constants as they can have consumers from other nGraph Function object
         const auto constant = ov::as_type_ptr<opset10::Constant>(node);
+        // skip precision sensitive marked Constants in ShapeOf subgraphs and leave them in f32 precision
+        if (constant && pass.transformation_callback(constant) && constant->get_output_element_type(0) == element::f32)
+            return res;
+
         const auto it = const_to_internal_output.find(node.get());
         if (constant && constant->get_output_element_type(0) == from && it != const_to_internal_output.end()) {
             return fuse_type_to_constant(node, to, it->second);
         }
 
         for (const auto& output : node->outputs()) {
+            // skip precision sensitive marked nodes in ShapeOf subgraphs and leave them in f32 precision
+            if (pass.transformation_callback(node) && output.get_element_type() == element::f32)
```

Comment:
same here

Reply:
see above

Reply:
Communicated with @itikhono and @jane-intel. Found a solution to remove these hardcoded checks from `ConvertPrecission`. Details are above.